test/unit/acknowledgement/process-invalid.test.js

```diff
@@ -0,0 +1,242 @@
+const mockCommit = jest.fn()
+const mockRollback = jest.fn()
+const mockTransaction = {
+  commit: mockCommit,
+  rollback: mockRollback
+}
+
+jest.mock('../../../app/data', () => {
+  return {
+    sequelize:
+       {
+         transaction: jest.fn().mockImplementation(() => {
+           return { ...mockTransaction }
+         })
+       }
+  }
+})
+
+jest.mock('../../../app/config')
+const mockConfig = require('../../../app/config')
+
+jest.mock('../../../app/event')
+const mockEvent = require('../../../app/event')
+
+jest.mock('../../../app/acknowledgement/get-hold-category-name')
+const getHoldCategoryName = require('../../../app/acknowledgement/get-hold-category-name')
+
+jest.mock('../../../app/holds/get-hold-category-id')
+const getHoldCategoryId = require('../../../app/holds/get-hold-category-id')
+
+jest.mock('../../../app/reschedule/index')
+const holdAndReschedule = require('../../../app/reschedule/index')
+
+jest.mock('../../../app/reset/index')
+const { resetPaymentRequestById } = require('../../../app/reset/index')
+
+const mockFRN = require('../../mock-frn')
+const processInvalid = require('../../../app/acknowledgement/process-invalid')
+const { SFI } = require('../../../app/schemes')
+
+let mockAcknowledgement
+let mockAcknowledgementError
+let mockHoldCategoryName
+let mockHoldCategoryId
+let mockHoldReschedule
+let mockResetPaymentRequestById
+
+let schemeId
+let paymentRequestId
+
+describe('send acknowledgement error event', () => {
+  beforeEach(() => {
+    mockConfig.isAlerting = true
+
+    mockAcknowledgement = JSON.parse(JSON.stringify(require('../../mock-acknowledgement')))
+    mockAcknowledgementError = JSON.parse(JSON.stringify(require('../../mock-acknowledgement-error')))
+    mockHoldCategoryName = JSON.parse(JSON.stringify(require('../../constants/hold-categories-names')))
+
+    schemeId = SFI
+    paymentRequestId = 1
+
+    getHoldCategoryName.mockReturnValue(
+      mockHoldCategoryName
+    )
+
+    getHoldCategoryId.mockReturnValue(
+      mockHoldCategoryId
+    )
+
+    holdAndReschedule.mockReturnValue({
+      mockHoldReschedule
+    })
+
+    resetPaymentRequestById.mockReturnValue({
+      mockResetPaymentRequestById
+    })
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should return undefined when processInvalid is called', async () => {
+    const result = await processInvalid(schemeId, paymentRequestId, mockFRN, mockAcknowledgementError)
+    expect(result).toBeUndefined()
+  })
+
+  test('should call resetPaymentRequestById when a schemeId, paymentRequestId, frn and transaction are given', async () => {
```

Comment:
`transaction` is not given, think it should be `acknowledgementError` -- applies to other test names too